src/3d/framegraph/qgs3daxisrenderview.h

```diff
@@ -0,0 +1,112 @@
+  Copyright            : (C) 2024 by Benoit De Mezzo
+  Email                : benoit dot de dot mezzo at oslandia dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DAXISRENDERVIEW_H
+#define QGS3DAXISRENDERVIEW_H
+
+#include "qgis_3d.h"
+#include "qgsabstractrenderview.h"
+#include <QObject>
+
+#define SIP_NO_FILE
+
+class QColor;
+class QRect;
+class QSurface;
+
+namespace Qt3DCore
+{
+  class QEntity;
+}
+
+namespace Qt3DRender
+{
+  class QRenderSettings;
+  class QCamera;
+  class QFrameGraphNode;
+  class QLayer;
+  class QViewport;
+  class QSubtreeEnabler;
+  class QRenderTargetSelector;
+} // namespace Qt3DRender
+
+class Qgs3DMapCanvas;
+class QgsCameraController;
+
+class QgsFrameGraph;
+class Qgs3DMapSettings;
+
+#define SIP_NO_FILE
+
+/**
+ * \ingroup 3d
+ * \brief 3d axis render view class
+ *
+ * \note Not available in Python bindings
+ * \since QGIS 3.40
+ */
+class _3D_EXPORT Qgs3DAxisRenderView : public QObject, public QgsAbstractRenderView
+{
+    Q_OBJECT
+  public:
+    /**
+     * Constructor for Qgs3DAxisRenderView with the specified \a parent object.
+     */
+    Qgs3DAxisRenderView( const QString &viewName, Qgs3DMapCanvas *canvas, //
+                         QgsCameraController *cameraCtrl, Qgs3DMapSettings *settings );
+
+    //! Returns the viewport associated to this renderview
+    Qt3DRender::QViewport *viewport() const;
+
+    //! Returns the layer to be used by entities to be included in the label renderpass
+    Qt3DRender::QLayer *labelLayer() const;
+
+    //! Returns main object layer
+    Qt3DRender::QLayer *objectLayer() const;
+
+    //! Returns main object camera (used for axis or cube)
+    Qt3DRender::QCamera *objectCamera() const;
+
+    //! Returns camera used for billboarded labels
+    Qt3DRender::QCamera *labelCamera() const;
+
+    virtual void updateWindowResize( int width, int height ) override;
+
+  public slots:
+    //! Updates viewport horizontal \a position
+    void onHorizontalPositionChanged( Qt::AnchorPoint position );
+
+    //! Updates viewport vertical \a position
+    void onVerticalPositionChanged( Qt::AnchorPoint position );
+
+    //! Updates viewport \a size. Uses canvas size by default.
+    void onViewportSizeUpdate( int width = -1, int height = -1 );
+
+  signals:
+    //! Emits new viewport scale factor when it change
+    void viewportScaleFactorChanged( double scaleFactor );
```

Comment:
this signal seems to be the only reason why this class is a QObject... and this signal if I understand correctly only needs to talk to Qgs3DAxis class. What do you think about this:
- this render view class will keep pointer to Qgs3DAxis object
- then viewport scale factor changes, it would do a direct function call on Qgs3DAxis rather than emitting signal
- the render view class can stop being a QObject, making things cleaner

Reply:
ok I will try that

Reply:
done!